advisor/src/main/kotlin/AdviceProvider.kt

```diff
@@ -43,7 +43,7 @@ abstract class AdviceProvider(val providerName: String) {
      * For a given list of [Package]s, retrieve findings and return a map that associates each package with a list of
      * [AdvisorResult]s. Needs to be implemented by child classes.
      */
-    abstract suspend fun retrievePackageFindings(packages: List<Package>): Map<Package, List<AdvisorResult>>
+    abstract suspend fun retrievePackageFindings(packages: Set<Package>): Map<Package, List<AdvisorResult>>
```

Comment:
Should the docs be updated? 
e.g. "For a given **set**..."

Reply:
Also, would it make sense to use `Collection` as type in function which are visible beyond class scope so that they can be called easily with any collection? (It would also simplify the caller code a bit (even as it is after applying the changes of this commit)

Reply:
One of the points of this change was to avoid all implementations to deduplicate the list before querying packages. Changing this to a generic collection would render that idea void. In general, if order and duplicates do not matter, I agree collection should be preferred over list / set. But here, I'd like to semantically express that implementation can rely on no duplicates being present.